packages/gatsby-source-shopify/src/lib.js

```diff
@@ -51,15 +51,16 @@ export const queryAll = async (
     ? aggregatedResponse.concat(nodes)
     : nodes
 
-  if (get([...path, `pageInfo`, `hasNextPage`], false, data))
-    return queryAll(
+  if (get([...path, `pageInfo`, `hasNextPage`], data)) {
+    return await queryAll(
```

Comment:
nit: I think these are equivalent (await vs return await)

Reply:
hah, you are right - I initially replaced return with await, only then to add return back thinking to limit changes that don't matter and forgetting I added await :D